src/logic/referral/errors.ts

```diff
@@ -27,3 +27,9 @@ export class SelfReferralError extends Error {
     super(`User cannot refer themselves: ${userAddress}`)
   }
 }
+
+export class ReferralEmailUpdateTooSoonError extends Error {
```

Comment:
If you'd like, you can extend this error to an `InvalidRequestError`, especially if we intend to respond to the request with a 400 status code. 

In that case, you would only need to catch that error in the handler and re-throw it.